src/Symfony/Component/HttpKernel/Controller/ControllerResolver.php

```diff
@@ -40,6 +47,14 @@ public function __construct(LoggerInterface $logger = null)
     }
 
     /**
+     * @internal
+     */
+    public function setArgumentResolverManager(ArgumentResolverManager $argumentResolverManager)
```

Comment:
Actually adding a new method to an `@api` class is also a BC break, you can move this to the constructor and allow a null which creates the default you currently have in `getArgumentResolverManager`

You can always overwrite the parameters of a constructor so there should be no BC breaks then :)
Plus you save you save method call.